Test Infrastructure: TypeScript errors and module resolution failures blocking coverage analysis

## Priority
🟡 **MEDIUM** - Blocks test coverage analysis and audit tasks

## Problem Summary
Running `npm test -- --coverage` reveals multiple test failures due to TypeScript errors and module resolution issues. These failures prevent accurate coverage analysis and indicate test infrastructure problems.

## Test Failures Detected

### 1. TypeScript Module Resolution Errors
Multiple test files cannot resolve `@principality/core`:

```
tests/integration/phase-4.2-mcp-pending-effects.test.ts:11:70
  error TS2307: Cannot find module '@principality/core' or its corresponding type declarations.

tests/unit/game-execute.test.ts:34:27
  error TS2307: Cannot find module '@principality/core'

tests/unit/action-card-parser-regression.test.ts:19:28
  error TS2307: Cannot find module '@principality/core'

tests/unit/game-end-detection.test.ts:32:28
  error TS2307: Cannot find module '@principality/core'
```

### 2. External Dependency Missing
```
tests/e2e/claude-api-helpers.ts:8:23
  error TS2307: Cannot find module '@anthropic-ai/sdk'
```

### 3. TypeScript Type Errors in E2E Tests
```
tests/e2e/haiku-gameplay.test.ts:569:7
  error TS1345: An expression of type 'void' cannot be tested for truthiness.
  
  expect(response).toContain('Province') || expect(response).toContain('end') ||
    expect(response).toContain('pile') || expect(response).toContain('score');
```

Invalid assertion chaining - `expect()` returns void, cannot use `||` operator.

```
tests/e2e/haiku-gameplay.test.ts:626:9
  error TS2554: Expected 2 arguments, but got 3.
  
  'Play with logging'
```

### 4. Test Assertion Failures
```
tests/unit/multiplayer-mcp-tools.test.ts:42:33
  expect(state.supply.size).toBe(8);
  
  Expected: 8
  Received: 17
```

**Two tests failing**: Supply size mismatch (expects 8, receives 17) in:
- UT 5.1: game_session - Create 2P Game
- UT 5.13: game_observe - Supply Info

## Test Run Summary
```
Test Suites: 8 failed, 15 passed, 23 total
Tests:       2 failed, 20 skipped, 551 passed, 573 total
```

## Impact on Audit
- **Cannot generate accurate coverage report** (Task 3 of Issue #19)
- **Cannot identify untested code paths**
- **Cannot complete test quality audit**

## Root Causes

### Module Resolution (@principality/core)
Likely causes:
1. **Missing workspace build**: Packages not built before test run
2. **tsconfig.json paths**: TypeScript path mapping not configured for tests
3. **jest.config.js**: Module resolution not configured correctly

### Supply Size Mismatch (8 vs 17)
Test expects 8 supply piles (Basic cards only?) but receives 17 (includes kingdom cards).

**Possible causes**:
- Test written for Phase 1 (no kingdom cards)
- Test not updated for Phase 4 (17 kingdom cards added)
- Game initialization changed but test expectations didn't

## Recommended Fixes

### Fix 1: Module Resolution
```bash
# Option A: Build packages first
npm run build

# Option B: Update jest.config.js to resolve workspace packages
moduleNameMapper: {
  '^@principality/(.*)$': '<rootDir>/..//src'
}
```

### Fix 2: TypeScript Assertion Chaining
```typescript
// BEFORE (invalid):
expect(response).toContain('Province') || expect(response).toContain('end');

// AFTER (valid):
const validResponses = ['Province', 'end', 'pile', 'score'];
expect(validResponses.some(str => response.includes(str))).toBe(true);

// OR use regex:
expect(response).toMatch(/Province|end|pile|score/);
```

### Fix 3: Supply Size Expectations
Update tests to expect correct supply size:
```typescript
// Update from Phase 1 expectations:
expect(state.supply.size).toBe(8);  // 7 basic + 1 kingdom?

// To Phase 4 expectations:
expect(state.supply.size).toBe(17); // 7 basic + 10 kingdom
```

## Investigation Needed
- [ ] Why did these tests pass in CI previously?
- [ ] Are tests running in wrong package context?
- [ ] Is there a pre-test build step missing?

## Discovered During
Audit for Issue #19 (Test Quality and Effectiveness) - Task 3: Coverage Gap Analysis

## Files Affected
- `tests/integration/phase-4.2-mcp-pending-effects.test.ts`
- `tests/unit/game-execute.test.ts`
- `tests/unit/action-card-parser-regression.test.ts`
- `tests/unit/game-end-detection.test.ts`
- `tests/e2e/claude-api-helpers.ts`
- `tests/e2e/haiku-gameplay.test.ts`
- `tests/unit/multiplayer-mcp-tools.test.ts`
- Jest/TypeScript configuration files

--- a/packages/cli/tsconfig.json
+++ b/packages/cli/tsconfig.json
@@ -13,7 +13,11 @@
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
-    "moduleResolution": "node"
+    "moduleResolution": "node",
+    "baseUrl": ".",
+    "paths": {
+      "@principality/core": ["../core/dist"]
+    }
   },
   "include": ["src/**/*"],
   "exclude": ["node_modules", "dist"]

--- a/packages/mcp-server/tsconfig.json
+++ b/packages/mcp-server/tsconfig.json
@@ -11,7 +11,11 @@
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "resolveJsonModule": true,
-    "moduleResolution": "node"
+    "moduleResolution": "node",
+    "baseUrl": ".",
+    "paths": {
+      "@principality/core": ["../core/dist"]
+    }
   },
   "include": ["src/**/*"],
   "exclude": ["node_modules", "dist", "tests"]

